c7n/resources/directory.py

```diff
@@ -109,6 +110,35 @@ def process_resource_set(self, client, directories, tags):
 Directory.filter_registry.register('marked-for-op', TagActionFilter)
 Directory.action_registry.register('mark-for-op', TagDelayedAction)
 
+@Directory.action_registry.register('delete')
+class DirectoryDelete(BaseAction):
+    """Delete a directory.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: delete-directory
+                resource: aws.directory
+                filters:
+                    - Name: test.example.com
+                actions:
+                  - delete
+    """
+    schema = type_schema('delete')
+    permissions = ('ds:DeleteDirectory',)
+
+    def process(self, resources):
+        client = local_session(
+            self.manager.session_factory).client('ds')
+
+        for resource_set in chunks(resources, size=100):
```

Comment:
Does the api only support a 100 resources per api call? I couldn't find it anywhere in the docs

Reply:
Good catch, will replace with normal for loop